constropt/constr_extractor/builtin_extractor.rb

```diff
@@ -242,117 +254,120 @@ def extract_builtin_numerical(ast)
     constraints
   end
 
-  def extract_builtin_has_one(ast)
+  def extract_builtin_foreign(ast)
+    constraints = []
     fields = []
     content = ast[1].children
-    foreign_key = nil
+    class_name = nil
+    fk_column_name = nil
     content.each do |node|
-      if node.type.to_s == "fcall"
-        next
-      end
       field = handle_symbol_literal_node(node)
       fields << field unless field.nil?
-      node.each do |n|
+      node.children.each do |n|
         k, v = handle_assoc_node(n)
-        if !k.nil? && k == 'class_name'
-          fields[0] = trim_string(v.source.to_s)
-        end
-        if !k.nil? && k == 'foreign_key'
-          foreign_key = trim_string(v.source.to_s)
-        end
+        fk_column_name = v if !k.nil? && (k == 'foreign_key')
+        class_name = v if !k.nil? && (k == 'class_name')
       end
     end
 
-    e = {"from_class" => @class_name, "class_name" => fields[0], "foreign_key" => foreign_key}
-    @has_ones.append(e)
-  end
-
-  def extract_builtin_has_many(ast)
-    fields = []
-    content = ast[1].children
-    foreign_key = nil
-    content.each do |node|
-      if node.type.to_s == "fcall"
-        next
-      end
-      field = handle_symbol_literal_node(node)
-      fields << field unless field.nil?
-      node.each do |n|
-        k, v = handle_assoc_node(n)
-        if !k.nil? && k == 'class_name'
-          fields[0] = trim_string(v.source.to_s)
-        end
-        if !k.nil? && k == 'foreign_key'
-          foreign_key = trim_string(v.source.to_s)
-        end
-      end
+    if class_name.nil?
+      class_name = fields[0].capitalize
```

Comment:
This is because if class_name or foreign_key is not specified, the ruby documentation I found here [https://guides.rubyonrails.org/association_basics.html](https://guides.rubyonrails.org/association_basics.html ) says that the default is based on the field name.

Reply:
There is only one field, but looping through all the nodes in content allows me to parse each node and at the same time extract the field